crates/compiler/build/src/llvm_passes.rs

```diff
@@ -0,0 +1,83 @@
+        },
+    }
+}
+
+pub fn optimize_llvm_ir(
+    env: &roc_gen_llvm::llvm::build::Env,
+    target: Target,
+    opt_level: OptLevel,
+    emit_debug_info: bool,
+    ll_file_path: &Path,
+) {
+    env.dibuilder.finalize();
+
+    if !emit_debug_info {
+        env.module.strip_debug_info();
+    }
+
+    // TODO: double check how much time is spent verifying here and below.
+    // For real compilation, we may not want to pay the cost.
```

Comment:
@bhansconnect do you think we need to do this before merging?

Reply:
no

Reply:
So I just did some quick test. Tested with a debug build of helloworld (cause that is the case where a little extra cost hurts the most).

The check here costs about 15ms. The second check is free. The total build time before linking is about 100ms. Linking without the surgical linker is about 150ms. My gut feeling is just to leave this for now with the todo and circle back later. Long term, we probably want to remove at least the first check here (when we do make it dump the llvm ir if the passes fail to run). But for now, verifying early seems like a good idea.